roqoqo/src/operations/analog_operations.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright © 2021-2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::operations::{
+    ImplementedIn1point10, InvolveQubits, InvolvedQubits, Operate, OperateSpinsAnalog, Substitute,
+    SupportedVersion,
+};
+use crate::RoqoqoError;
+use qoqo_calculator::CalculatorFloat;
+use std::collections::{HashMap, HashSet};
+use struqture::spins::{PauliProduct, SpinHamiltonian};
+use struqture::OperateOnDensityMatrix;
+use struqture::SpinIndex;
+
+/// Implements the continuous time, constant spin Hamiltonian
+#[derive(Debug, Clone, PartialEq, roqoqo_derive::Operate, roqoqo_derive::Substitute)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "json_schema", derive(schemars::JsonSchema))]
+pub struct ApplyConstantSpinHamiltonian {
+    /// Hamiltonian to be simulated.
+    pub hamiltonian: SpinHamiltonian,
+    /// Time of evolution.
+    pub time: CalculatorFloat,
+}
+
+#[allow(non_upper_case_globals)]
+const TAGS_ApplyConstantSpinHamiltonian: &[&str; 4] = &[
+    "Operation",
+    "ModeGateOperation",
+    "OperateSpinsAnalog",
```

Comment:
This should be `SpinsAnalogOperation`, you'll need to add the enum in the macros. You can base this on the `ModeGateOperation` enum.